### PR TITLE
Updates the promote post behavior for sync in progress error

### PR DESCRIPTION
**Pull request by sbarbosa**
When you first connect Jetpack to WPCOM, there is a delay for the post to get synced. During that time, if you access the Blaze Dashboard, you will see a blocking error preventing you from creating a campaign until the site finishes sinking. 

The main problem is that the user can't promote old posts because we won't have information about them in WPCOM, but new posts will be ready to promote as soon as they are created. 

To improve this error, we are changing the message to be a warning and not blocking the result of the `/blaze/posts` endpoint.

## Proposed Changes

* Adds a behavior to handle warning messages from the `/blaze/posts` endpoint

## Why are these changes being made?

* We want to improve the first contact of the users with the dashboard, when they are using an external site with Jetapck or any other Jetpack-connected plugin.

## Testing Instructions

Jetpack still sends an error on that endpoint, so the only way to test this before we merged a change in Jetpack is by hardcoding the response of the endpoint (or changing it on a sandbox).

I will mention the step for the hardcode approach, but feel free to sandbox the public-api and change the `/blaze/posts` endpoint to start returning the new `warnings` prop.

* Checkout this branch and start the Server
* Edit this file and change the line 91 to:
```
warnings: [ 'sync_in_progress' ],
```
* Navigate to Tools->Advertising
* Verify that you see the new Info message and the full list of posts.

Screenshot 2024-09-24 at 5 03 04 PM

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by therocket-gr — APPROVED** [review]
works as described
code changes LGTM
Screenshot 2024-09-26 at 12 24 13

my version has a pink checkmark (as it uses a different color theme) and looks a bit more important than yours.. not sure if we should just hardcode the bg color to always have it blue??

**Comment by sbarbosa** [comment]
Thanks for the review @therocket-gr! 

> not sure if we should just hardcode the bg color to always have it blue??

Yes, I'm not sure either. It's currently using the default styling for the notices in WPCOM. I think it's OK to follow the standard and show the notice like the rest of the admin panel. But this is just my opinion. 

Let me know if you want to apply a change here.

File: client/my-sites/promote-post-i2/main.tsx [hunk 1]
```diff
@@ -228,6 +232,33 @@ export default function PromotedPosts( { tab }: Props ) {
 		);
 	};
 
+	const renderWarningNotices = ( warnings?: string[] ) => {
+		const content = [];
+
+		for ( const item of warnings ?? [] ) {
+			switch ( item ) {
+				case 'sync_in_progress':
```
Comment (j6ll) [inline client/my-sites/promote-post-i2/main.tsx:241]: Wonder if we should store this as a const somewhere?

**Review by j6ll — APPROVED** [review]
Thanks for adding this Seba, works as described 👍 

Left a minor comment, but looks good otherwise :)

Screenshot 2024-09-26 at 16 38 55

File: client/my-sites/promote-post-i2/main.tsx (hunk 1 above)
Comment (sbarbosa) [reply client/my-sites/promote-post-i2/main.tsx:241]: I could add some typescript support to it. Let me create a type for the warning list 👍

Comment (sbarbosa) [reply client/my-sites/promote-post-i2/main.tsx:241]: Done in this commit: https://github.com/Automattic/wp-calypso/pull/94862/commits/e5737a544077a988fba886d89393de71fe6d0291

Now, the IDE will suggest the values, and we will get a warning if we try to use it without the correct type/value.

**Pull request merged**